tasmota/tasmota_xsns_sensor/xsns_115_wooliis.ino

```diff
@@ -0,0 +1,230 @@
+  char float_str[8];
+  if (Wooliis.valid) {
+    if (json) {
+      ResponseAppend_P(PSTR(",\"%s\":{"), D_WOOLIIS);
+      dtostrfd(Wooliis.remaining_capacity, 1, float_str);
+      ResponseAppend_P(PSTR("\"%s\":%s,"), D_CAPACITY, float_str);
+      ResponseAppend_P(PSTR("\"%s\":%d,"), D_CHARGING, (Wooliis.status ? 1 : 0));
+      dtostrfd(Wooliis.voltage, 1, float_str);
+      ResponseAppend_P(PSTR("\"%s\":%s,"), D_VOLTAGE, float_str);
+      dtostrfd(Wooliis.current, 1, float_str);
+      ResponseAppend_P(PSTR("\"%s\":%s,"), D_CURRENT, float_str);
+      dtostrfd(Wooliis.voltage*Wooliis.current, 1, float_str);
+      ResponseAppend_P(PSTR("\"%s\":%s,"), D_POWERUSAGE, float_str);
+      dtostrfd(Wooliis.energy_out, 1, float_str);
+      ResponseAppend_P(PSTR("\"%s\":%s"), D_ENERGY, float_str);
+      dtostrfd(Wooliis.energy_in, 1, float_str);
+      ResponseAppend_P(PSTR("\"%s\":%s,"), D_ENERGY "_returned", float_str);
+      ResponseJsonEnd();
+#ifdef USE_DOMOTICZ
+      if (0 == TasmotaGlobal.tele_period) {
+        char pe[16];
+        snprintf_P(pe, sizeof(pe), PSTR("%d;%d"), Wooliis.voltage*Wooliis.current, Wooliis.energy_out);
+        DomoticzSensor(DZ_COUNT, Wooliis.charge_percent);
+        DomoticzSensor(DZ_VOLTAGE, Wooliis.voltage);
+        DomoticzSensor(DZ_CURRENT, Wooliis.current);
+        DomoticzSensor(DZ_POWER_ENERGY, pe);
+      }
+#endif  // USE_DOMOTICZ
+    }   // if json
+#ifdef USE_WEBSERVER
+    else {
+      WSContentSend_PD(HTTP_SNS_WOOLIIS_BCP, D_BATTERY, Wooliis.charge_percent);
+      dtostrfd(Wooliis.remaining_capacity, 1, float_str);
+      WSContentSend_PD(HTTP_SNS_WOOLIIS_CAP, D_BATTERY, float_str);
+      WSContentSend_PD(HTTP_SNS_WOOLIIS_CHG, D_BATTERY, (Wooliis.status ? D_TRUE : D_FALSE));
+      dtostrfd(Wooliis.voltage, 1, float_str);
+      WSContentSend_PD(HTTP_SNS_WOOLIIS_VOL, D_BATTERY, float_str);
+      dtostrfd(Wooliis.current, 1, float_str);
+      WSContentSend_PD(HTTP_SNS_WOOLIIS_CUR, D_BATTERY, float_str);
+      dtostrfd(Wooliis.voltage*Wooliis.current, 1, float_str);
+      WSContentSend_PD(HTTP_SNS_WOOLIIS_POW, D_BATTERY, float_str);
+      dtostrfd(Wooliis.energy_in, 1, float_str);
+      WSContentSend_PD(HTTP_SNS_WOOLIIS_EIN, D_BATTERY, float_str);
+      dtostrfd(Wooliis.energy_out, 1, float_str);
+      WSContentSend_PD(HTTP_SNS_WOOLIIS_EEG, D_BATTERY, float_str);
+    }
+#endif  // USE_WEBSERVER
+  }
+}
+
+/*********************************************************************************************\
+ * Interface
+\*********************************************************************************************/
+
+bool Xsns115(uint32_t function)
+{
+  bool result = false;
+
+  switch (function) {
+    case FUNC_EVERY_SECOND:
+      WooliisSecond();
+      break;
+    case FUNC_JSON_APPEND:
+      WooliisShow(1);
+      break;
+#ifdef USE_WEBSERVER
+    case FUNC_WEB_SENSOR:
+      WooliisShow(0);
+      break;
+#endif  // USE_WEBSERVER
+    case FUNC_INIT:
+      WooliisInit();
```

Comment:
Move this up like in xsns_28_qmp6988.ino. The goal is to abandon Xsns115 ASAP if not configured. So the test for ``Wooliis.valid`` would become into the else part of the if (FUNC_INIT == ..)